addons/mail/models/mail_thread.py

```diff
@@ -401,6 +406,14 @@ def _creation_subtype(self):
         """
         return self.env['mail.message.subtype']
 
+    def _creation_subtype_message(self):
```

Comment:
As we now have 3 utility methods used at create_time (``_creation_subtype``, ``_creation_message`` and now ``_creation_subtype_message``), don't you think we should have a single utility method returning a dict with all necessary information ? Especially I think that there is some overlap between the two 'message' methods: either you have a subtype (with its message), either not (with another message). Can't we have a method returning subtype and message directly ? Something like
```
def _creation_post_data(self):
    return {
        'subtype_id': self.env['mail.message.subtype'],
        'body': _('%(model_name)s created',
                  model_name=self.env['ir.model']._get(self._name).name),
    }
```

And potential overrides would look like
```
def _creation_post_data(self):
    if self.foo == 'bar':
        return {
            'subtype_id': self.envref('module.subtype_ref'),
            'body': _('A new foo has been created in container %(container_name)s',
                      container_name=self.container_id.name,
        }
    return super()._creation_post_data()
```
Probably to test, but it would simplify the custom implementation of mail.thread, as you will have only 1 method to define instead of 3 :)